src/main/java/org/araymond/joal/core/ttorrent/client/Client.java

```diff
@@ -61,161 +71,169 @@ void setAnnouncerExecutor(final AnnouncerExecutor announcerExecutor) {
     public void start() {
         this.stop = false;
 
+        // TODO: use @Scheduled or something similar instead of looping manually over X period
         this.thread = new Thread(() -> {
             while (!this.stop) {
-                for (final AnnounceRequest req : this.delayQueue.getAvailables()) {
+                this.delayQueue.getAvailables().forEach(req -> {
                     this.announcerExecutor.execute(req);
                     try {
                         this.lock.writeLock().lock();
-                        this.currentlySeedingAnnouncer.removeIf(an -> an.equals(req.getAnnouncer())); // remove the last recorded event
-                        this.currentlySeedingAnnouncer.add(req.getAnnouncer());
+                        // TODO: do we need removeIf()? why not use remove(req.getAnnouncer())? because list may contain more instances of same announcer?
```

Comment:
Would like to understand why are we using `removeIf()` here

Reply:
Should be se same indeed